File: src/BaekJoon/C++/baekjoon_10828.cpp
// https://www.acmicpc.net/problem/10828

#include <iostream>
#include <string>
using namespace std;

class stackNode {
public :
	int data;
	stackNode *next;

	stackNode() {
		data = 0;
		next = NULL;
	}

	stackNode(int data, stackNode *next) {
		this->data = data;
		(*this).next = next;
	}
};

class stack {
private :
	stackNode *topNode;
	int stackSize;
public :
	stack() {
		topNode = NULL;
		stackSize = 0;
	}

	void push(int);

	int pop() {
		if(empty()) {
			return -1;
		}

		int temp;

		temp = topNode->data;
		topNode = topNode->next;
		stackSize--;

		return temp;

	}

	int empty();
	int top() {
		if(empty())
			return -1;
		else
			return topNode->data;
	}

	int size();
};

void stack::push(int data) {
	topNode = new stackNode(data, topNode);
	stackSize++;
}

int stack::empty() {
	if(!topNode)
		return 1;
	else
		return 0;
}

int stack::size() {
	return stackSize;
}

int main() {
	int n;
	stack s;

	cin >> n;

	for(int i = 0; i < n; i++) {
		string str;
		cin >> str;

		if(str == "push") {
			int x;
			cin >> x;
			s.push(x);
		} else if(str == "pop") {
			cout << s.pop() << "\n";
		} else if(str == "size") {
			cout << s.size() << "\n";
		} else if(str == "empty") {
			cout << s.empty() << "\n";
		} else if(str == "top") {
			cout << s.top() << "\n";
		}
	} // end of for

	return 0;
}

File: src/BaekJoon/Java/baekjoon_10828.java
package BaekJoon.Java;
// https://www.acmicpc.net/problem/10828

import java.util.Scanner;

class Stack_Node {

	Stack_Node next;
	int data;

	Stack_Node() {
		next = null;
		data = 0;
	}

	Stack_Node(int element) {
		this.data = element;
	}

	Stack_Node(int element, Stack_Node next) {
		this.data = element;
		this.next = next;
	}
}

class Stack {
	protected Stack_Node top;
	protected int size = 0;

	Stack() {
		top = null;
	}

	void push(int x) {
		size++;
		top = new Stack_Node(x, top);
	}

	int pop() {
		if(size == 0)
			return -1;

		size--;

		int element = top.data;
		top = top.next;

		return element;
	}

	int empty() {
		if(size == 0)
			return 1;

		return 0;
	}

	int top() {
		if(size == 0)
			return -1;

		return top.data;
	}
}

public class baekjoon_10828 {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		Stack stack = new Stack();

		for(int test_case = 1; test_case <= n; test_case++) {
			String s = sc.next();

			if(s.equals("push")) {
				int x = sc.nextInt();

				stack.push(x);
			} else if(s.equals("pop"))
				System.out.println(stack.pop());
			else if(s.equals("size"))
				System.out.println(stack.size);
			else if(s.equals("empty"))
				System.out.println(stack.empty());
			else if(s.equals("top"))
				System.out.println(stack.top());
		}

		sc.close();
	}
}